### ISSUE TITLE
[SYCL] Ambiguous reference to `PointerType`

### ISSUE THREAD

**Issue by 0x12CC** [intel/llvm#21734]
### Describe the bug

563401d956cd33d845618cecc1ef1ba32d51e0ec adds a new definition for `PointerType` in `llvm/include/llvm/SYCLLowerIR/MangleUtils.h` that conflicts with the one in `llvm/include/llvm/IR/InlineAsm.h`. Although the two definitions are in different namespaces, the `MangleUtils.cpp` file has using directives for both namespaces.

There's no error when compiling with GCC or with pre-compiled headers disabled since `MangleUtils.cpp` does not include `InlineAsm.h`. The error only occurs when compiling with pre-compiled headers enabled since they include the conflicting definition.

### To reproduce

Compile using Clang with pre-compiled headers enabled. Compilation should produce the following diagnostic:
```
llvm/lib/SYCLLowerIR/MangleUtils.cpp:100:6: error: reference to 'PointerType' is ambiguous
```

### Environment

- OS: Linux
- Target device and vendor: NA
- DPC++ version: 563401d956cd33d845618cecc1ef1ba32d51e0ec
- Dependencies version: NA

**Comment by 0x12CC** [comment]
@wenju-he, friendly ping.

**Comment by wenju-he** [comment]
@0x12CC will be fixed in https://github.com/intel/llvm/pull/21739
